### PR TITLE
Maintenance Part 4: Adding Maintenance Status To Linode Table

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Adds "Maintenance Status" Column to Linodes Landing.

## Type of Change
- Non breaking change ('update', 'change')

## Applicable E2E Tests

N/A

## Note to Reviewers

Use prod-test-003 with the following test data:

```js
{
  "results": 5,
  "page": 1,
  "data": [
    {
      "until": null,
      "entity": {
        "url": "/linode/instances/500",
        "id": 14341983,
        "type": "linode",
        "label": "loadbal1-vagrant"
      },
      "when": null,
      "body": null,
      "severity": "major",
      "type": "migration_scheduled",
      "label": "You have a scheduled migration pending!",
      "message": "You have a scheduled migration pending!  Your Linode will be automatically shut down, migrated, and returned to its last state."
    },
    {
      "until": null,
      "entity": {
        "url": "/linode/instances/504",
        "id": 14341983,
        "type": "linode",
        "label": "loadbal5-xengrant"
      },
      "when": null,
      "body": null,
      "severity": "major",
      "type": "migration_pending",
      "label": "You have a migration pending!",
      "message": "You have a migration pending! Your Linode must be offline before starting the migration."
    },
    {
      "until": null,
      "entity": {
        "url": "/linode/instances/500",
        "id": 14341983,
        "type": "linode",
        "label": "loadbal1-vagrant"
      },
      "when": "2019-06-29T15:32:26",
      "body": "This Linode resides on a host that is pending critical maintenance. You should have recieved a support ticket that details how you will be affected. Please see the aformentioned ticket and [status.linode.com](https://status.linode.com/) for more details.",
      "severity": "critical",
      "type": "maintenance",
      "label": "zombieload-migration-scheduled",
      "message": "This Linode will be affected by critical maintenance!"
    },
    {
      "until": null,
      "entity": {
        "url": "/linode/instances/500",
        "id": 14341983,
        "type": "linode",
        "label": "loadbal1-vagrant"
      },
      "when": "2019-10-29T15:32:26",
      "body": "This Linode resides on a host that is pending critical maintenance. You should have recieved a support ticket that details how you will be affected. Please see the aformentioned ticket and [status.linode.com](https://status.linode.com/) for more details.",
      "severity": "critical",
      "type": "maintenance",
      "label": "zombieload-migration-scheduled",
      "message": "This Linode will be affected by critical maintenance!"
    },
    {
      "until": null,
      "entity": {
        "url": "/linode/instances/500",
        "id": 14341983,
        "type": "linode",
        "label": "loadbal1-vagrant"
      },
      "when": "2019-02-28T15:32:26",
      "body": "This Linode resides on a host that is pending critical maintenance. You should have recieved a support ticket that details how you will be affected. Please see the aformentioned ticket and [status.linode.com](https://status.linode.com/) for more details.",
      "severity": "critical",
      "type": "maintenance",
      "label": "zombieload-migration-scheduled",
      "message": "This Linode will be affected by critical maintenance!"
    },
    {
      "until": null,
      "entity": {
        "url": "/linode/instances/507",
        "id": 14341983,
        "type": "linode",
        "label": "loadbal8-xengrant"
      },
      "when": "2020-03-22T18:58:41",
      "body": "This Linode resides on a host that is pending critical maintenance. You should have recieved a support ticket that details how you will be affected. Please see the aformentioned ticket and [status.linode.com](https://status.linode.com/) for more details.",
      "severity": "critical",
      "type": "maintenance",
      "label": "zombieload-reboot-scheduled",
      "message": "This Linode will be affected by critical maintenance!"
    },
    {
      "until": null,
      "entity": {
        "url": "/linode/instances/507",
        "id": 14341983,
        "type": "linode",
        "label": "loadbal8-xengrant"
      },
      "when": "2019-03-22T18:58:41",
      "body": "This Linode resides on a host that is pending critical maintenance. You should have recieved a support ticket that details how you will be affected. Please see the aformentioned ticket and [status.linode.com](https://status.linode.com/) for more details.",
      "severity": "critical",
      "type": "maintenance",
      "label": "zombieload-reboot-scheduled",
      "message": "This Linode will be affected by critical maintenance!"
    }
  ],
  "pages": 1
}
```

### CODE REVIEW

File: src/features/linodes/LinodesLanding/LinodeRow/LinodeRow.tsx [hunk 1]
```diff
@@ -136,6 +140,15 @@ export const LinodeRow: React.StatelessComponent<CombinedProps> = props => {
             <IPAddress ips={ipv4} copyRight showCopyOnHover />
           </div>
         </TableCell>
+        <TableCell
+          parentColumn="Maintenance Status"
+          className={classes.regionCell}
+          data-qa-maintenance-status
+        >
+          {maintenanceStartTime
+            ? formatDate(maintenanceStartTime)
```
Comment (martinmckenna) [inline src/features/linodes/LinodesLanding/LinodeRow/LinodeRow.tsx]: lots of error handling coming in the next PR. With tests and reporting to sentry if this returns `Invalid Date`

File: src/features/linodes/LinodesLanding/SortableTableHead.tsx [hunk 2]
```diff
@@ -60,6 +60,14 @@ const SortableTableHead: React.StatelessComponent<combinedProps> = props => {
         >
           IP Address
         </TableSortCell>
+        <TableSortCell
+          label="maintenance:when" // we want to sort by the first ipv4
```
Comment (martinmckenna) [inline src/features/linodes/LinodesLanding/SortableTableHead.tsx]: remove comment

File: src/components/OrderBy.ts [hunk 3]
```diff
@@ -51,9 +51,24 @@ export const sortData = (orderBy: string, order: Order) =>
...
+     * @todo document this in a README
+     *
+     * this allows us to pass a value such as 'maintenance:when' to the handleOrderChagne
+     * callback and it will turn it into a pathOr-friendly format
+     *
+     * so "maintenance:when" turns into ['maintenance', 'when']
+     *
+     * useful for when you want to sort by a nested property
+     */
+    if (orderBy.includes(':')) {
```
Comment (Jskobos) [inline src/components/OrderBy.ts]: Does this work? Same idea but less complicated:
```{js}
    const _orderBy = orderBy.split(':');

    /** basically, if orderByProp exists, do a pathOr with that instead */
    const aValue = pathOr('', _orderBy, a);
    const bValue = pathOr('', _orderBy, b);
```

Comment (martinmckenna) [reply src/components/OrderBy.ts]: there's some additional logic a few lines up that will make that impl. difficult.

**Pull request merged**